WL4EditorWindow.cpp

```diff
@@ -2457,3 +2460,25 @@ void WL4EditorWindow::on_spinBox_RoomID_valueChanged(int arg1)
         if (sender() != nullptr) SetCurrentRoomId(arg1, true);
     }
 }
+
+void WL4EditorWindow::on_action_swap_Rooms_triggered()
+{
+    unsigned int currentroomid = ui->spinBox_RoomID->value();
+    bool okay = false;
+    int value = QInputDialog::getInt(this, QApplication::applicationName(),
+                                     tr("input a Room Id to apply the Room swap with the current Room.\n"
+                                        "Notice: You CANNOT undo this step !"), currentroomid,
+                                     0, CurrentLevel->GetRooms().size() - 1, 1, &okay);
+    if (okay && value != currentroomid)
```

Comment:
Let's print a warning or make a popup warning that you cannot swap the same room with itself (value == currentroomid)